### PR TITLE
User signup details

### PR DESCRIPTION
**Pull request by sbsmith86**
#### What's this PR do?
APOLOGIES IN ADVANCES FOR THIS LARGE PR. I GOT REFACTOR HAPPY. 

**CHANGES**
---
* Adds a `show()` method to the `SignupController` to generate an empty view that will hold individual signup information. [here](https://github.com/DoSomething/rogue/compare/user-signup-details?expand=1#diff-28d9d92df597ce839f7f0cfc3a8dfb94R47)
*Links `SignupCards` to their signup view. [here](https://github.com/DoSomething/rogue/compare/user-signup-details?expand=1#diff-ff380d564bdc83de86dff5ffd7dcd30bR25)

Creates some helper components: 
  * [`MetaInformation`](https://github.com/DoSomething/rogue/compare/user-signup-details?expand=1#diff-8680797a7b96a051028288bc8c1cfd44R4) - can be used to show meta information throughout the app. I saw this popping up a lot 
  * [`TextBlock`](https://github.com/DoSomething/rogue/compare/user-signup-details?expand=1#diff-ab69fd96c88368341b685c1e5a5ec33bR3) - Creates a block of text with a bold headline and text under it.
  * [`UserInformation`](https://github.com/DoSomething/rogue/compare/user-signup-details?expand=1#diff-02b1fe65556b371bcdb00cceba274d71R5) - Used to show a block of user information and can take in other elements/components to display under it.
  * [`Quantity`](https://github.com/DoSomething/rogue/compare/user-signup-details?expand=1#diff-d46977e27a6c7274ed2e507f97807097R3) - Used to show the fancy box of quantity with the noun and verb next to it. 
  * [`ReviewBlock`](https://github.com/DoSomething/rogue/compare/user-signup-details?expand=1#diff-18d307dbdc33d56a9696c2923ef00b20R6) - A component to hold all of the review interactive elements.
  * And the [`Signup`](https://github.com/DoSomething/rogue/compare/user-signup-details?expand=1#diff-01916e56b863fd54af704a818a7f14cfR35) component which is the container component for the whole page. 

Updates some functionality: 
 * Updated the `Tags` component to put an `...` indicator when something is being tagged. 
 * Updated the `HistoryModal` so that it only takes in the Signup it is editing and the campaign tied to the signup. 
    - NOTE: This component could use some more work. It should be able to take in a post as well. I removed that requirement for now because right now we are only editing the signup with it and eventually it will need to be able to either edit the signup AND show history for it or edit the Signup and show history for the post. Worth looking at this functionality as a whole at somepoint
* Updated the `CampaignInbox` and `CampaignSingle` components to work with the updates. 

#### How should this be reviewed?

I would take a look at the notes above and check out each of the changes individually instead of trying to look at it all at once. I did a lot of back and forth so my commit history is a little wonky (apologies!)

#### Any background context you want to provide?

I am hoping that some of these components will help us clean up other places around the app, like the `InboxItem` component. 

Also I have made tickets to refactor the `InboxItem` component and to look at how we decided which container component to add to the page so we don't have to keep adding if-statements to `app.js`. See [150273936
](https://www.pivotaltracker.com/story/show/150273936) and [150272139](https://www.pivotaltracker.com/story/show/150272139)

Also I will be doing more clean up work once these new components get merged in. 

#### Relevant tickets
https://www.pivotaltracker.com/story/show/150173770

#### Checklist
- [ ] Documentation added for new features/changed endpoints.
- [ ] Tested on staging.
- [ ] Pinged a PM if this is a larger PR that would benefit from some additional testing love.

### CODE REVIEW

File: resources/assets/components/MetaInformation/index.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import React from 'react';
+import { map } from 'lodash';
+
+const MetaInformation = (props) => (
+  <div className="container">
+    {props.title ? <h4 className="heading">{props.title}</h4> : null}
+    {
+      map(props.details, (item, key) => {
```
Comment (chloealee) [inline resources/assets/components/MetaInformation/index.js]: Love this! Super smart extraction that works for all types of information.

File: resources/assets/components/UserOverview/index.js [hunk 2]
```diff
@@ -61,21 +63,16 @@ class UserOverview extends React.Component {
...
-            {cityState ? <span>{cityState}<br/></span> : null }
-          </p>
-        </div>
-        <div className="container__block">
-          <h4 className="heading">Meta</h4>
-          <p>
-            <span>Source: {user.source}<br/></span>
-            <span>Northstar ID: {user.id}<br/></span>
-          </p>
+          <UserInformation user={user} includeMeta={false}>
```
Comment (chloealee) [inline resources/assets/components/UserOverview/index.js]: @sbsmith86 what does `includeMeta` do here?

Comment (sbsmith86) [inline resources/assets/components/UserOverview/index.js]: Oh great catch! it does nothing, something left over from a previous iteration of the code.

**Comment by chloealee** [comment]
@sbsmith86 besides the merge conflicts, all looks good to me!

File: app/Http/Controllers/SignupsController.php [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+{
+    /**
+     * Registrar instance
+     *
+     * @var Rogue\Services\Registrar
+     */
+    protected $registrar;
+
+    /**
+     * Phoenix instance
```
Comment (DFurnes) [inline app/Http/Controllers/SignupsController.php]: 👾

File: app/Http/Controllers/SignupsController.php [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+    /**
+     * Display the specified resource.
+     *
+     * @param  string  $id  signup id
+     * @return \Illuminate\Http\Response
+     */
+    public function show($id)
+    {
+        $signup = Signup::with('posts')->find($id);
```
Comment (DFurnes) [inline app/Http/Controllers/SignupsController.php]: Do we want to `findOrFail` here so this 404s if the signup doesn't exist?

File: resources/assets/components/MetaInformation/index.js (hunk 1 above)
Comment (DFurnes) [inline resources/assets/components/MetaInformation/index.js]: Agreed! 🏆

**Pull request merged**